packages/devtools-core/src/utils/communication.ts

```diff
@@ -0,0 +1,41 @@
+import { Dispatch } from "@prodo/core/lib/types";
+import { state } from "../model";
+import { Action, AppMessage, DevMessage } from "../types";
+
+export const recordState = (newState: any) => {
+  state.app.state = newState;
+};
+
+export const recordAction = (action: Action) => {
+  state.app.actionLog.push(action);
+  if (action.rerender && Object.keys(action.rerender).length > 0) {
+    Object.keys(action.rerender).forEach(rerender => {
+      const renderLogEntry = {
+        componentId: rerender,
+        actionName: action.actionName,
+      };
+      state.app.renderLog.push(renderLogEntry);
```

Comment:
I may be wrong about this, but I think the idea was to show the number of times each component has re-rendered, as opposed to just a list of components re-rendering.

Reply:
Ah, sure. So more like this:

List1: rendered 2 times
Item10: rendered 12 times
...

In alphabetical order?

Reply:
Probably don't need the number either. The name is what is important to the user.

But I think that is the general idea. We might want to have it expandable and show a count for which actions triggered a rerender

List: rendered 10 times
  - 8 triggered by action1
  - 2 triggered by action2

Reply:
True... It might be easier to leave as is for now, get this merged, and then my next task will be to figure out a more useful render tab?